test/unit_tests/container/generic_data_handle.cpp

```diff
@@ -0,0 +1,124 @@
+#include "neuron/container/generic_data_handle.hpp"
+#include "neuron/model_data.hpp"
+
+// Pull `Datum`.
+#include "neuron/cache/model_data.hpp"
+
+#include <catch2/catch.hpp>
+
+using namespace neuron::container;
+
+
+TEST_CASE("POINTER raw pointer") {
+    double* v = (double*) malloc(sizeof(double));
+    double* w = (double*) malloc(sizeof(double));
+
+    double* const v_copy = v;
+    double* const w_copy = w;
+
+    // generic_data_handle v_gdh(v); // Not implemented.
+    Datum v_gdh;
+    v_gdh = v;
+    REQUIRE(v_gdh.get<double*>() == v);
+
+    // Attempt 1a: Assign `w` via `generic_data_handle.get<double*>()`. This is
+    // bound to fail, because the pointer is returned by value. Hence this
+    // assigns to an rvalue.
+    // v_gdh.get<double*>() = w;
+    //
+    // Results in a straight up compilation error:
+    //     test/unit_tests/container/generic_data_handle.cpp:23:21: error: lvalue required as left
+    //     operand of assignment
+    //     23 |   v_gdh.get<double*>() = w;
+    //        |   ~~~~~~~~~~~~~~~~~~^~
+    //
+    // Attempt 1b: Store the rvalue.
+    double* tmp = v_gdh.get<double*>();
+    tmp = w;
+
+    REQUIRE(v_gdh.get<double*>() == v);
+    REQUIRE(v_gdh.get<double*>() != w);
```

Comment:
The context is how do we define `_p_c` in generated code. Here the particular concern is code what would like to assign a pointer to `_p_c`. Something, loosely, modulo some `void *` casts:
```
_p_c = &printf
```
and then later (add appropriate casting):
```
_p_c("yay");
```

Obviously, this can't work. That's the only point. The comment that makes me think this might be relevant is:
https://github.com/neuronsimulator/nrn/issues/2458#issuecomment-1681131074